src/Rings/MPolyMap/flattenings.jl

```diff
@@ -35,6 +37,30 @@
                                                     )
   end
 
+  function RingFlattening(
```

Comment:
There's a lot of code duplication here for the graded cases. But I didn't find how to avoid this, really. Maybe the people involved in the development of graded stuff have some ideas?